tests/providers/amazon/aws/hooks/test_batch_client.py

```diff
@@ -0,0 +1,314 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import unittest
+
+import botocore.exceptions
+from parameterized import parameterized
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.batch_client import AwsBatchClient
+from tests.compat import mock
+
+# clear any existing AWS credentials
+AWS_REGION = "eu-west-1"
+os.environ["AWS_DEFAULT_REGION"] = AWS_REGION
+os.environ["AWS_REGION"] = AWS_REGION
```

Comment:
This is definitely a no-go. Please use mock in tests that requires environment variables:
```
@unittest.mock.patch.dict('os.environ', AIRFLOW__CORE__DEFAULT_IMPERSONATION=TEST_USER)
```

Reply:
And this is the reason behind failed tests. This code is performed during tests collection thus creates side effects that break AWS test.